### PR TITLE
[TFF-145] Checkin model

### PR DESCRIPTION
**Pull request by tondef001**
### Summary

Ability to check in a checked out model

### Checklist

- [x] Have you added unit and functional tests as appropriate?
- [x] Did you update/add documentation for new methods or changed functionality?
- [x] Have you merged the target branch down into this one?

### How to Review

1. Get the address of a model available for checkout (such as: m8y7nFLsT)
2. Go to the student page and check out a large number of that model
3. First, just click the "Check in" Button on the student panel. A single model should be checked in with accompanying toast.
4. You can use the arrows on the number input to change the number checked in at once.  Entering numbers also works, and any number greater than the number checked out or less than 1 will be changed to the max or 1 respectively.
5. Clicking check in all will unsurprisingly check in the rest of the model, regardless of how many there are.
6.  Play around with checking in/out models in different amounts.  Periodically check the model page to make sure the quantity is being updated appropriately.
7. Run tests and review code

### Links

- [TFF-145](https://msoese.atlassian.net/browse/TFF-145)
- TheFourFifths/consus/pull/72

### CODE REVIEW

File: test/functional/model-checkin.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+            method: 'get',
+            endpoint: 'model/all',
+            response: {
+                status: 'success',
+                data: {
+                    models
+                }
+            }
+        });
+        return app.client.keys('999999').then(() => {
```
Comment (gearedmonkey) [inline test/functional/model-checkin.js]: Why does this work? The server is expecting a get request with query string 123456 no 99999

File: src/controllers/pages/student.js [hunk 2]
```diff
@@ -47,6 +47,19 @@ export default class StudentController{
...
+        return checkInModel(id, modelAddress, quantity).then(data => {
+            return searchStudent(id).then(student => {
+                Dispatcher.handleAction('MODEL_CHECKIN_SUCCESS', {
+                    modelAddress: data.modelAddress,
+                    modelName: data.modelName,
+                    quantity: data.quantity
+                });
+                Dispatcher.handleAction("STUDENT_FOUND", student);
+            });
+        });
```
Comment (gearedmonkey) [inline src/controllers/pages/student.js]: Need a .catch at the end of this promise for when the server fails to do the checkin.

File: test/unit/controllers/pages/student.js [hunk 3]
```diff
@@ -127,6 +127,53 @@ describe("StudentController", () => {
         });
     });
 
+    describe("checkInModel", () => {
```
Comment (gearedmonkey) [inline test/unit/controllers/pages/student.js]: Make sure to add test for the checkInModel returning a rejected promise after adding the error catch to the controller.

File: docs/Actions.md [hunk 4]
```diff
@@ -27,6 +27,7 @@ This document describes the Flux actions used in the Consus client.
     - [ITEM_FOUND](#item_found)
     - [MODELS_RECEIVED](#models_received)
     - [MODEL_CREATED](#model_created)
+    - [MODEL_CHECKIN_SUCCESS](#model_checkin_success)
```
Comment (thofmann) [inline docs/Actions.md]: :+1:

File: test/functional/item-checkout.js [hunk 5]
```diff
@@ -77,10 +77,10 @@ describe('Checking an item out', function () {
             qs: {
                 address: 'iGwEZUvfA'
             },
-            response:{
+            response: {
```
Comment (thofmann) [inline test/functional/item-checkout.js]: :)

**Comment by thofmann** [comment]
I think there will be merge conflicts when we try merging the story into `dev`.

**Pull request merged**